src/twisted/application/runner/test/test_runner.py

```diff
@@ -86,12 +106,23 @@ def test_runUsesPIDFile(self):
         """
         L{Runner.run} uses the provided PID file.
         """
-        pidFile = DummyPIDFile()
 
-        runner = DummyRunner(pidFile=pidFile)
+        def __enter__(self):
+            self.entered = True
+            return self
+
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            self.exited = True
+
+        self.patch(NonePIDFile, "__enter__", __enter__)
+        self.patch(NonePIDFile, "__exit__", __exit__)
 
-        self.assertFalse(pidFile.entered)
-        self.assertFalse(pidFile.exited)
+        pidFile = NonePIDFile()
```

Comment:
Rather than abusing `NonePIDFile` here, at this point couldn't you just write a little in-line `@contextmanager` that set a flag, and pass that, skipping the `.patch` calls?